x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,574 @@
+the existing PKCS#12 file (such as `elastic-certificates.p12`) with the new
+certificate. Stop one node at a time, replace the node certificate, and then
+restart the node.
+
+NOTE: The following examples use PKCS#12 files, but the same steps apply to JKS
+keystores.
+
+. Open the `ES_PATH_CONF/elasticsearch.yml` file and check the names of the
+keystores that are currently in use. You'll use the same names for your new
+certificate.
++
+In this example, the keystore and truststore are pointing to different files.
+Your configuration might use the same file that contains the certificate and CA.
+In this case, include the path to that file for both the keystore and truststore.
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. On every node in your cluster, generate a certificate and private key for your
+node. You must include the CA that was used to sign the certificate currently
+in use.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `--ca <ca_file>`:: Name of the CA keystore used to sign your certificates. The
+   default file name from the `elasticsearch-certutil` tool is
+   `elastic-stack-ca.p12`.
+====
++
+    a. Enter the password for your CA keystore when prompted.
+
+    b. Enter the name for your keystore, which must match the name of the
+    keystore currently in use.
++
+NOTE: If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename. 
+
+    c. (Optional) Enter a password for your keystore. 
+
+. If you entered a password when creating the node certificate, run the
+following command to store the password in the {es} keystore:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.keystore.secure_password
+----
+--
+
+. Stop {es} on the first node in your cluster where you are updating the
+keystore.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `elastic-stack-ca.p12`.
+
+. If your keystore is password protected, the password is stored in the {es}
+keystore, _and_ the password needs to change, then update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore
+file that you created.
+
+. Restart the node where you updated the keystore. 
+
+. (Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. On each additional node in your cluster, stop {es}, update the keystore,
+and restart {es}.
+
+[[node-certs-same-http]]
+==== Generate a new certificate for the HTTP layer
+The certificate for the HTTP layer is used to verify other components when they
```

Comment:
the other components verify this certificate when connecting to {es}, it's not {es} that verifies other components when they connect

Reply:
I'll change to read:
>Other components such as {kib} or any of the Elastic language clients verify this certificate when they connect to {es}.